storage/filesystem/internal/dotgit/dotgit.go

```diff
@@ -103,3 +106,70 @@ func (d *DotGit) Idxfile() (fs.FS, string, error) {
 
 	return nil, "", ErrIdxNotFound
 }
+
+// Objectfiles returns the hash slice of the objects files in the "objects/"
+func (d *DotGit) Objectfiles() (fs.FS, []core.Hash, error) {
+	objsDir := d.fs.Join(d.path, "objects")
+
+	files, err := d.fs.ReadDir(objsDir)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var objects []core.Hash
+	for _, f := range files {
+		if f.IsDir() && len(f.Name()) == 2 && isHex(f.Name()) {
+			objDir := f.Name()
+			objs, err := d.fs.ReadDir(d.fs.Join(objsDir, objDir))
+			if err != nil {
+				return nil, nil, err
+			}
+
+			for _, obj := range objs {
+				objects = append(objects, core.NewHash(objDir+obj.Name()))
+			}
+		}
+	}
+
+	return d.fs, objects, nil
+}
+
+func isHex(s string) bool {
+	for _, b := range []byte(s) {
+		if isNum(b) {
+			continue
+		}
+		if isHexAlpha(b) {
+			continue
+		}
+
+		return false
+	}
+
+	return true
+}
+
+func isNum(b byte) bool {
+	return b >= '0' && b <= '9'
+}
+
+func isHexAlpha(b byte) bool {
+	return b >= 'a' && b <= 'f' || b >= 'A' && b <= 'F'
+}
+
+// Objectfile returns the path of the object file (really, it returns the
+// path of the first file in the "objects/hash[0:2]/"
```

Comment:
Can you be a little more specific about what this function does? Example: `Objectfile returns the path of the object file for a given hash *if the file exists*, otherwise returns an ErrObjfileNotFound error.`

Please remove the `(really, it returns the path of the first file in the "objects/hash[0:2]/"`, *it is wrong*, badly formatted (a `)` is missing) and it is a leftover from a too evident copy and paste operation.